pkg/registry/core/service/ipallocator/allocator.go

```diff
@@ -263,8 +263,8 @@ func calculateIPOffset(base *big.Int, ip net.IP) int {
 // RangeSize returns the size of a range in valid addresses.
 func RangeSize(subnet *net.IPNet) int64 {
 	ones, bits := subnet.Mask.Size()
-	if (bits - ones) >= 31 {
-		panic("masks greater than 31 bits are not supported")
+	if bits == 32 && (bits-ones) >= 31 || bits == 128 && (bits-ones) >= 63 {
```

Comment:
As a future clean-up, we should look at making this easier to understand (or supplement it with a comment to indicate that if the network bits are less than 2 for IPv4 or less than 65 for IPv6 it fails. OK for now.